src/components/RegistrationTable/data-table.tsx

```diff
@@ -1,94 +1,116 @@
-"use client"
+"use client";
 
-import { useState, useRef, useEffect } from "react"
-import { DataTableProps, SortingState, ColumnFiltersState } from "./types"
-import { useReactTable, getCoreRowModel, getPaginationRowModel, getSortedRowModel, getFilteredRowModel } from "@tanstack/react-table"
-import { TableComponent, TableHeader } from "./TableComponent"
-import { TableHeader as TableActionsHeader } from "./TableHeader"
-import { RegistrationsTableBody } from "./RegistrationsTableBody"
-import { TableFooter } from "./TableFooter"
-import { useColumnVisibility } from "./hooks/useColumnVisibility"
-import { columns as defaultColumns, Attendee } from "./columns"
+import { useState, useRef, useEffect } from "react";
+import { DataTableProps, SortingState, ColumnFiltersState } from "./types";
+import {
+  useReactTable,
+  getCoreRowModel,
+  getPaginationRowModel,
+  getSortedRowModel,
+  getFilteredRowModel,
+} from "@tanstack/react-table";
+import { TableComponent, TableHeader } from "./TableComponent";
+import { TableHeader as TableActionsHeader } from "./TableHeader";
+import { RegistrationsTableBody } from "./RegistrationsTableBody";
+import { TableFooter } from "./TableFooter";
+import { useColumnVisibility } from "./hooks/useColumnVisibility";
+import { columns as defaultColumns, Attendee } from "./columns";
+import QrCheckIn from "../QrScanner/QrScanner";
 
-export function DataTable({ initialData, dynamicColumns = [] }: DataTableProps<Attendee>) {
-    const [data, setData] = useState(initialData)
-    const [sorting, setSorting] = useState<SortingState>([])
-    const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([])
-    const [rowSelection, setRowSelection] = useState<Record<string, boolean>>({})
-    const [pageSize, setPageSize] = useState(10)
-    const [isClient, setIsClient] = useState(false)
+export function DataTable({
+  initialData,
+  dynamicColumns = [],
+  eventId,
+  year,
+}: DataTableProps<Attendee>) {
+  const [data, setData] = useState(initialData);
+  const [sorting, setSorting] = useState<SortingState>([]);
+  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([]);
+  const [rowSelection, setRowSelection] = useState<Record<string, boolean>>({});
+  const [pageSize, setPageSize] = useState(10);
+  const [isClient, setIsClient] = useState(false);
+  const [qr, setQr] = useState(false);
```

Comment:
I agree with making this boolean a hook in the parent component, but not sure about the name. At a glance, it's a bit hard to tell what the hook is for.

I think we could improve the readability of this hook by renaming it to something like  `isQrReaderToggled, setQrReaderToggled`. This would make it more clear that this is a boolean, and that it's meant to toggle the QR Reader. What do you think? This link describes the reasoning behind it under the sections 'boolean-naming' and 'hook-use-state' https://www.codiga.io/blog/enforce-react-best-practices/

This naming change would also apply to renaming the component props

Reply:
ty for feedback!